src/actions/sagemaker/training_job_poller.ts

```diff
@@ -0,0 +1,159 @@
+/* tslint:disable max-line-length */
+import * as SageMaker from "aws-sdk/clients/sagemaker"
+import * as nodemailer from "nodemailer"
+import * as winston from "winston"
+import * as Hub from "../../hub"
+import { getMailTransporter } from "./mail_transporter"
+import { logRejection } from "./utils"
+
+export const FIVE_MINUTES = 1000 * 60 * 5
+export const THIRTY_SECONDS = 1000 * 30
+
+export interface Transaction {
+  request: Hub.ActionRequest
+  sagemaker: SageMaker
+  modelName: string
+  jobName: string
+  roleArn: string
+  trainingImage: string
+  maxRuntimeInSeconds: number
+  pollIntervalInSeconds: number
+}
+
+export class TrainingJobPoller {
+
+  transporter: nodemailer.Transporter
+  intervalTimer: any
+  timeoutTimer: any
+
+  constructor(transaction: Transaction) {
+    this.transporter = getMailTransporter(transaction.request)
+    this.pollTrainingJob(transaction).catch(logRejection)
+  }
+
+  async pollTrainingJob(transaction: Transaction) {
+      // start poller for training job completion
+      winston.debug("starting poller")
+
+      this.intervalTimer = setInterval(() => {
+        this.checkTrainingJob(transaction).catch(logRejection)
```

Comment:
Got a lot of lint errors about not handling promises correctly. As a fix I added these .catch handlers to log the rejection. Not sure if this is an adequate solution.